inspirehep/modules/records/receivers.py

```diff
@@ -98,6 +119,20 @@ def assign_uuid(sender, *args, **kwargs):
             author['uuid'] = str(uuid.uuid4())
 
 
+@before_record_insert.connect
+@before_record_update.connect
+def update_collections(sender, *args, **kwargs):
```

Comment:
Currently, you are only adding to `_collections`, never removing.

I think the behavior should be the following:
- if a record has `special_collections`, it should get the corresponding `_collections` without `Literature`.
- if a record has no `special_collections`, it should get `Literature` in `_special_collections`.

There is still the issue of how to assign stuff having both `Literature` and a special collection.

Reply:
This is implementing only the behavior of [`copy_special_collections`](https://github.com/inspirehep/inspire-dojson/blob/8c02690ea5d997b46c8dc48e937f6017ddc05176/inspire_dojson/hep/model.py#L85-L90), because it assumes that whatever is in `_collections` is already correct... I think that this commit is correct, but maybe we need to discuss this in person.